File: problems/009/009.go
package main

import "fmt"
import "os"
import "strconv"
import "time"

func f() uint64 {
    perimeter := uint64(1000)

    for c := perimeter / 3 + 1; c < perimeter / 2; c++ {
        for b := (perimeter - c) / 2 + 1; b < c; b++ {
            a := perimeter - b - c

            if a * a + b * b == c * c {
                return a * b * c
            }
        }
    }

    return 0
}

func main() {
    if len(os.Args) == 1 {
        fmt.Println(f())
    } else if len(os.Args) == 2 {
        iters, _ := strconv.ParseInt(os.Args[1], 10, 64)

        start := time.Now()
        for i := int64(0); i < iters; i++ {
            f()
        }
        end := time.Now()

        fmt.Println(end.Sub(start).Nanoseconds())
    }
}

File: problems/009/009.rs
extern crate test;
extern crate time;

use std::os::args;
use test::black_box;
use time::precise_time_ns;

static PERIMETER: u64 = 1000;

#[inline]
fn f() -> u64 {
    for c in range(PERIMETER / 3 + 1, PERIMETER / 2) {
        for b in range((PERIMETER - c) / 2 + 1, c) {
            let a = PERIMETER - b - c;

            if a * a + b * b == c * c {
                return a * b * c
            }
        }
    }

    unreachable!();
}

fn main() {
    match args().as_slice() {
        [_] => {
            println!("{}", f());
        },
        [_, ref iters] => {
            let iters: u64 = from_str(iters.as_slice()).unwrap();

            let start = precise_time_ns();
            for _ in range(0, iters) {
                black_box(f());
            }
            let end = precise_time_ns();

            println!("{}", end - start);
        },
        _ => unreachable!(),
    }
}
